libbeat/outputs/shipper/config.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package shipper
+
+import (
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/common/transport/tlscommon"
+
+	"google.golang.org/grpc/backoff"
+)
+
+type Backoff struct {
```

Comment:
Nit: we have backoff implementation already in beats: https://github.com/elastic/beats/tree/main/libbeat/common/backoff
I would prefer if we could reuse it or adjust that to fit this use case.

Reply:
what do you mean? it's an internal implementation in gRPC, I don't think we can re-use the one we have.

Reply:
Well, gRPC has its own backoff parameters, but when we expose backoff of an input or output in Beats we have established field names etc, and something that's exposed as a beats configuration object should probably follow those and convert to the gRPC API internally.

In this case though, I think I'd rather remove the gRPC backoff settings entirely and just use reasonable defaults? I don't think there's much to be gained by having ways to fine-tune backoff jitter at this stage -- we're a ways off from any mass deployments and it seems good to keep configurations minimal while we're still figuring out where the libraries connect.